### DISCUSSION TITLE
🔥Dynamic Routes in Server Routes for Nuxt 3🔥

### DISCUSSION THREAD

**Discussion by sync42johnny** [nuxt/nuxt#22444] [Questions]
I'm currently exploring dynamic routing in Nuxt 3 and I came across the documentation that explains how to define dynamic routes in the pages directory. According to the documentation here: https://nuxt.com/docs/guide/directory-structure/pages#dynamic-routes, dynamic routes can be easily defined in the pages directory using the file and directory names. For example:
```
-| pages/
---| index.vue
---| users-[group]/
-----| [id].vue
```
I'm particularly intrigued by the users-[group]/ part of the structure, which seems to suggest that dynamic parameters can be used in parent directory names as well. This could greatly simplify the routing logic for some of my use cases.

My question is this: Does Nuxt 3 support the same functionality for server routes? Can I dynamically generate server routes based on parameters like in the users-[group]/ example?

If so, could you kindly guide me on how to implement this? If not, are there plans to include this feature in future releases, or could you recommend a workaround?

Your insights would be greatly appreciated!

Thank you.😀

**Comment by lazercaveman** [comment]
[According to the Nuxt Docs](https://nuxt.com/docs/guide/directory-structure/server#server-routes) "currently server routes do not support the full functionality of dynamic routes as [pages](https://nuxt.com/docs/guide/directory-structure/pages#dynamic-routes) do."

But could be a good thing to implement in future releases I think - to handle dynamic sitemaps, rss-feeds, and things like that...
